### ISSUE TITLE
Detect coroutines with no await / async operations

### ISSUE THREAD

**Issue by cooperlees** [cooperlees/flake8-async#2]
Async function like below `receive` has no await statements. It makes `foo` running forever without giving back control to the event loop.

```python
import asyncio

loop = asyncio.get_event_loop()

async def foo():
    async def receive():  # should warning or info, this can be a sync function
        return "haha"

    while True:
        data = await receive()


async def bar():
    while True:
        await asyncio.sleep(1)
        print("Hello")


loop.create_task(foo())
loop.create_task(bar())
loop.run_forever()
```

**Comment by Zac-HD** [comment]
This feels like a special case of "must await on every code path, or none", so how would you feel about picking up a bunch of `trio`-related checks?  ([including e.g. this wishlist](https://gitter.im/python-trio/general?at=629ff3c206a77e1e18766216))

I'm keen to put them *somewhere*, just a question about the scope of this package, and your feelings on the balance between "only alert when we're pretty sure it's a bug" to "alert on any unidiomatic or suspicious code".

**Comment by cooperlees** [comment]
I thought you were thinking of a `flake8-trio` dedicated plugin ...

I'm open to make this cover all async frameworks and have a type of error for each async framework that a check makes sense for ... e.g. check foo - asyncio + trio, check bar - twisted etc.

This project basically has 0 usage now so we still have time to design it more.

**Comment by Zac-HD** [comment]
I was, but there are *some* checks that make sense for both (which it would be a shame to write twice) as well as plenty that are framework-specific.

I think on balance this comes down to whether we can sufficiently-reliably detect whether asyncio or trio is being used; I suspect we can do that from imports but if that's not enough then we should probably implement separate plugins.  I'll aim to spend a while sketching out what an "all-in-one" plugin would look like sometime in the next week or two...

**Comment by cooperlees** [comment]
This sounds like a good plan. Happy to see the sketch, but going to need another person to help this. I won't have the cycles to drive it out by myself. Happy to transfer ownership etc.

**Comment by Zac-HD** [comment]
I've gone ahead and created https://github.com/Zac-HD/flake8-trio/ to hold trio-specific checks, since they're basically disjoint idioms.  My proposal is then that everything here should be relevant to *any* async framework, and if it picks up asyncio-specific checks those should either be in a different plugin, or gated behind a check that asyncio is imported into the module.

**Comment by jakkdl** [comment]
This is now implemented in flake8-trio with TRIO107 and TRIO108, and afaik should work for this context as well.
